src/index.js

```diff
@@ -96,7 +102,7 @@ module.exports = (options) => {
     };
 
     if (json && !decodeBody) {
-        throw new Error('If parseBody is set to true the decodeBody must be also true.');
+        throw new Error('If json is set to true the decodeBody must be also true.');
```

Comment:
This will throw synchronously while all the other rejects are async. We should not mix sync and async. Making the function async will fix the problem.

Reply:
fixed